codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala

```diff
@@ -478,9 +477,23 @@ object ModelBuilder {
       fqnExtractor: FullyQualifiedNameExtractor): FullyQualifiedName = {
 
     val (revolvedPackage, resolvedName) = extractPackageAndName(pkg, name)
-    val descriptor = lookupDescriptor(revolvedPackage, additionalDescriptors)
+    val descriptor = lookupDescriptor(revolvedPackage, resolvedName, additionalDescriptors)
     resolveFullyQualifiedMessageType(resolvedName, descriptor, additionalDescriptors)
   }
+  private def resolveTypeArgument(name: String, pkg: String, additionalDescriptors: Seq[Descriptors.FileDescriptor])(
```

Comment:
We need this extra method when resolving scalar types for replicated entities.
Instead of `resolveFullyQualifiedMessageType` and then building a `TypeArgument`, we must build a `TypeArgument` directly.